Author: A. Jesse Jiryu Davis <jesse@mongodb.com>
Description: Apply upstream patch to fix CDRIVER-2014
Index: libbson/Makefile.am
===================================================================
--- libbson.orig/Makefile.am
+++ libbson/Makefile.am
@@ -44,16 +44,22 @@ EXTRA_DIST += CMakeLists.txt \
     VERSION_RELEASED
 
 dist-hook:
-	@if test -d "$(srcdir)/.git"; then                              \
-          (cd "$(srcdir)" &&                                            \
-           $(top_srcdir)/build/autotools/missing --run git log --stat ) > ChangeLog.tmp \
-           && mv -f ChangeLog.tmp $(top_distdir)/ChangeLog              \
-           || (rm -f ChangeLog.tmp;                                     \
-               echo Failed to generate ChangeLog >&2);                  \
-	else                                                            \
-	  echo A git checkout is required to generate a ChangeLog >&2;  \
+	-rm -rf $(distdir)/doc/html/.buildinfo $(distdir)/doc/html/.doctrees
+	-rm -rf $(distdir)/doc/man/.buildinfo $(distdir)/doc/man/.doctrees
+
+	@if test -d "$(srcdir)/.git"; then                                   \
+	       (cd "$(srcdir)" &&                                            \
+	        $(top_srcdir)/build/autotools/missing --run git log --stat ) > ChangeLog.tmp \
+	        && mv -f ChangeLog.tmp $(top_distdir)/ChangeLog              \
+	        || (rm -f ChangeLog.tmp;                                     \
+	            echo Failed to generate ChangeLog >&2);                  \
+	else                                                                 \
+	  echo A git checkout is required to generate a ChangeLog >&2;       \
 	fi
 
+clean-local:
+	rm -rf $(CLEANFILES)
+
 uninstall-local:
 	-rmdir $(bsondocdir)/doc $(bsondocdir)/html
 	-rm -r $(bsondocdir)
Index: libbson/doc/Makefile.am
===================================================================
--- libbson.orig/doc/Makefile.am
+++ libbson/doc/Makefile.am
@@ -9,9 +9,3 @@ EXTRA_DIST += \
   doc/mongoc-theme/static/pygments.css \
   doc/mongoc-theme/theme.conf \
   doc/taglist.py
-
-dist-hook: man html
-
-clean-local:
-	-rm -rf doc/man/.buildinfo doc/man/.doctrees
-	-rm -rf doc/html/.buildinfo doc/html/.doctrees doc/html/_static
Index: libbson/doc/man/Makefile.am
===================================================================
--- libbson.orig/doc/man/Makefile.am
+++ libbson/doc/man/Makefile.am
@@ -1,19 +1,14 @@
-BUILT_MAN_FILES = $(wildcard doc/man/*.3)
-CLEANFILES += $(wildcard doc/man/*.3)
+CLEANFILES += $(wildcard doc/man/*.3) doc/man/.doctrees doc/man/.buildinfo
 
-dist_man_MANS = $(BUILT_MAN_FILES)
-# Automake (at least up to 1.10) mishandles dist_man_MANS inside conditionals.
-# Unlike with other dist primaries, the files are not distributed if the
-# conditional is false.
-# Work the bug around until it is fixed:
-dist_noinst_DATA = $(dist_man_MANS)
+.PHONY: doc/man
 
-man3_MANS = $(BUILT_MAN_FILES)
+dist_man_MANS = $(wildcard doc/man/*.3)
 
-.PHONY: man
+man3_MANS = $(wildcard doc/man/*.3)
 
-# "E" is "ignore cache", "b" is build type.
-man:
+
+# sphinx-build "-E" is "ignore cache", "b" is build type.
+doc/man:
 	-mkdir -p doc/man
 	$(SPHINX_BUILD) -qEW -b man $(top_srcdir)/doc doc/man
 
