Author: A. Jesse Jiryu Davis <jesse@mongodb.com>
Description: Apply upstream patch to fix CDRIVER-2014
Index: libbson/doc/html/Makefile.am
===================================================================
--- libbson.orig/doc/html/Makefile.am
+++ libbson/doc/html/Makefile.am
@@ -19,5 +19,6 @@ doc/html: $(SPHINX_HTML_FILES)
 $(SPHINX_HTML_FILES):
 	-mkdir -p doc/html
 	$(SPHINX_BUILD) -qEW $(top_srcdir)/doc doc/html
+	rm -rf doc/html/.buildinfo doc/html/.doctrees
 
 EXTRA_DIST += doc/html
Index: libbson/doc/man/Makefile.am
===================================================================
--- libbson.orig/doc/man/Makefile.am
+++ libbson/doc/man/Makefile.am
@@ -11,5 +11,6 @@ man3_MANS = $(wildcard doc/man/*.3)
 doc/man:
 	-mkdir -p doc/man
 	$(SPHINX_BUILD) -qEW -b man $(top_srcdir)/doc doc/man
+	rm -rf doc/man/.buildinfo doc/man/.doctrees
 
 EXTRA_DIST += doc/man
Index: libbson/Makefile.am
===================================================================
--- libbson.orig/Makefile.am
+++ libbson/Makefile.am
@@ -59,8 +59,6 @@ dist-hook:
 
 clean-local:
 	-rm -rf $(CLEANFILES)
-	-rm -rf doc/html/.buildinfo doc/html/.doctrees
-	-rm -rf doc/man/.buildinfo doc/man/.doctrees
 
 uninstall-local:
 	-rmdir $(bsondocdir)/doc $(bsondocdir)/html
